src/esp/sensor/SensorFactory.cpp

```diff
@@ -12,7 +12,21 @@ sensor::SensorSuite SensorFactory::createSensors(
   sensor::SensorSuite sensorSuite = sensor::SensorSuite();
   for (const sensor::SensorSpec::ptr& spec : sensorSetup) {
     scene::SceneNode& sensorNode = node.createChild();
-    sensorSuite.add(sensor::CameraSensor::create(sensorNode, spec));
+    // VisualSensor Setup
```

Comment:
Is this a good way to check and initialize sensors? I moved sensorType and sensorSubType out of SensorSpec, so now casting is the only way to tell the type. Would it be better to move sensorType and sensorSubType back in since all sensors will define this?

Reply:
> I moved sensorType and sensorSubType out of SensorSpec

Why are you doing this? They should be in the SensorSpec.

Reply:
Good to know it was a step in the wrong direction. Based on this comment it seemed we wanted to remove it so I tried it out, I will put it back.  (https://github.com/facebookresearch/habitat-sim/pull/1090#discussion_r575594474).